perceval/backends/core/googlehits.py

```diff
@@ -174,34 +174,30 @@ def __parse_hits(self, hit_raw):
 
         # Strip the hits
         fetched_on = datetime_utcnow().timestamp()
+        hits_json = {
+            'fetched_on': fetched_on,
+            'id': self.__uuid(self.keywords[:], fetched_on),
+            'keywords': self.keywords,
+            'type': 'googleSearchHits'
+        }
+
         if not hit_string:
             logger.warning("No hits for %s", self.keywords)
-            hits_json = {
-                'fetched_on': fetched_on,
-                'hits': 0,
-                'id': self.__uuid(*self.keywords),
-                'keywords': self.keywords,
-                'type': 'googleSearchHits'
-            }
+            hits_json['hits'] = 0
 
             return hits_json
 
         str_hits = re.search('\d+', hit_string).group(0)
         hits = int(str_hits)
-        hits_json = {
-            'fetched_on': fetched_on,
-            'hits': hits,
-            'id': self.__uuid(*self.keywords),
-            'keywords': self.keywords,
-            'type': 'googleSearchHits'
-        }
+        hits_json['hits'] = hits
 
         return hits_json
 
     @staticmethod
-    def __uuid(*args):
+    def __uuid(args, fetched_on):
         """Generate a UUID based on the given parameters."""
 
+        args.append(str(fetched_on))
         s = '-'.join(args)
```

Comment:
On `supybot` backend we don't use any characters to concatenate strings when we generate the `id`. Not sure what it's better.

Reply:
I took as reference https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backend.py#L444

Reply:
this is gone, now uuids rely on the method `uuid` in backend.py